sale_coupon_auto_refresh/models/sale_order.py

```diff
@@ -49,19 +49,25 @@ class SaleOrderLine(models.Model):
 
     @api.model_create_multi
     def create(self, vals_list):
-        lines = super().create(vals_list)
+        lines = super(
+            SaleOrderLine, self.with_context(skip_auto_refresh_coupons=True)
+        ).create(vals_list)
         lines.mapped("order_id")._auto_refresh_coupons()
         return lines
 
     def write(self, vals):
         orders = self.mapped("order_id")
-        res = super().write(vals)
+        res = super(
+            SaleOrderLine, self.with_context(skip_auto_refresh_coupons=True)
+        ).write(vals)
         orders |= self.mapped("order_id")
         orders._auto_refresh_coupons()
```

Comment:
One issue we're experiencing is the performance of this. Maybe we should restrict the recomputations to a list of safe fields, like `product_id`, `product_uom_qty`, `product_uom_id`. `price_unit`, `discount`. This way, we'd avoid unnecessary updates.

The same would go for the sale order.